### ISSUE TITLE
Additional Banner features

### ISSUE THREAD

**Issue by briand44** [uiowa/uids#949]
# Story

As a web team, we would like to add additional features to the banner component.

# Additional Info

Epic https://github.com/uiowa/uids/issues/721

https://github.com/uiowa/uids/issues/721#issuecomment-1217096616

- **Alignment options:** 
  - Look at https://github.com/uiowa/uids/issues/721#issuecomment-1217096616
  - Switching this up could allow us to introduce `Top` alignment. 
- **Aspect ratio:** 
  - Vertical, Widescreen, Ultra Widescreen? How would this differ from the sizes?  Vertical would be most useful to help showcase portrait images. 
  - Maybe an option (for desktop probably, but possibly for mobile) that allows `100vh` - height of the header?
- https://github.com/uiowa/uids/issues/968 
- **Padding options:**
  - The majority of the banner CSS in `uids_base` is altering the padding based on the layout container: https://github.com/uiowa/uiowa/blob/main/docroot/themes/custom/uids_base/scss/components/banner.scss.  For this, look into setting small, medium, large padding controls that will adjust through container queries. 
- ~Replace CTA component~
- **Refactor CSS to reduce complexity:**
  - Look at the background color nesting
    - A lot of chaining and nesting is occurring because when the `[Mobile] Content Below Image/Video` is selected using dark/light gradient it has to adapt to the container background color for mobile. 
      - A possible fix for this would be to set a fixed background color for dark/light gradients so that it would not have to adapt to the parent container background color and be a fixed black or white value. 
    - Another set of chaining and nesting is occurring because we are not removing the gradient when a background color is selected.  The gradient classes should only be applied on Image/Video and not when a background color is used.  Setting form logic and an update hook in `uids_base` to remove/show/hide these options/classes only when Image/Video is selected would reduce more CSS complexity.

**Comment by e-marie-w** [comment]
Grooming convo: do work in both uids and uiowa as part of task. Replace CTA component - in UIDS means folding it into the banner. In uiowa, could rework template/style sheet of CTA block to use banner stuff. Decided to split into two issues - separate CTA changes from rest of updates. Other features first, then CTA because alignment options are needed/affected by CTA.

**Comment by pyrello** [comment]
Follow up issue created for CTA work at #958.

**Comment by briand44** [comment]
blocked until https://github.com/uiowa/uids/issues/721

**Comment by pyrello** [comment]
In #950, there was code added to the `background.ts` file with the full list of background options:
https://github.com/uiowa/uids/blob/f18d05e05f29ab088eea048c68b77545fdae99a0/src/components/shared/background.ts#L11-L28

That could be used to supply the options that were added in this commit: https://github.com/uiowa/uids/pull/950/commits/e191661d4fd1c1380674db8963a841e72283f5c9.

**Comment by pyrello** [comment]
> Refactor CSS to reduce complexity

tl;dr I think there is more to do here and its important to keep going on this now.

This CSS work was deferred from the first slice to avoid any complexity related to possibly having to refactor the component template in SiteNow. However, this is an important component of this work as it also is very much part of the 4.x banner conversion and re-launch. I've been taking a look at this with an eye towards reducing complexity and I think there is a lot more to be done here. There is a lot of cruft and rules are complex because it was updated as we went. There is a lot of copy/paste weirdness (the usage throughout of selectors like `[class*="bg--gray"]` instead of `.bg--gray`).

I think its important to do this before we finish this and move on because we'll be dealing with the choices made for years to come. We have a tendency to look back at our previous choices non-critically and to give deference to our previous choices even when some of them weren't well-reasoned or even made intentionally. Finishing the cleanup now will help us avoid a future where we are stuck with dozens of duplicative rules canceling other rules because nobody has any idea why we made them but we assume they were important and rewriting them is too hard.

**Comment by pyrello** [comment]
Some notes about progress so far:
- @bspeare and I were able to refactor the SCSS to remove the need for classes related to gradient orientation. This allows us to avoid having to add those classes in code in the .vue file and in SiteNow.
- There were a lot of conflicting rules setting the background for the `.banner:after` pseudo element. I made updates to set (mobile-first) base rules and removed rules that don't change those base rules.
- When we worked on the card updates, we made a deliberate choice to move away from such a deeply nested CSS structure, instead favoring more top-level rules directly referencing the element being modified. I have started transitioning the banner SCSS similarly and this is still in progress.
- I have reorganized the file to follow this progression:
  - Banner base rules
  - Banner element (e.g. headline, text) rules
  - Modifier (e.g. horizontal alignment, gradient color) rules
- We currently still have a lot of modifier rules mixed in with our element rules, which I think makes it harder to see the total picture of what a modifier is doing. My plan is to refactor so that most/all modifier rules for a particular modifier are grouped together.
